File: app/src/main/java/de/markusfisch/android/shadereditor/adapter/ShaderSpinnerAdapter.java
package de.markusfisch.android.shadereditor.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SpinnerAdapter;

public class ShaderSpinnerAdapter
		extends ShaderAdapter
		implements SpinnerAdapter {
	/**
	 * Constructs a new adapter for a spinner.
	 *
	 * @param context The current context.
	 */
	public ShaderSpinnerAdapter(Context context) {
		// Call the parent constructor, which only requires a Context.
		super(context);
	}

	/**
	 * This method is required by the SpinnerAdapter interface to display
	 * the items in the dropdown list.
	 *
	 * @param position    The position of the item within the adapter's data set.
	 * @param convertView The old view to reuse, if possible.
	 * @param parent      The parent that this view will eventually be attached to.
	 * @return A View that displays the data at the specified position.
	 */
	@Override
	public View getDropDownView(
			int position,
			View convertView,
			ViewGroup parent) {
		// We can reuse the parent's getView() implementation because the
		// layout for a selected item and a dropdown item is the same.
		return getView(position, convertView, parent);
	}
}

File: app/src/main/java/de/markusfisch/android/shadereditor/adapter/ShaderSpinnerAdapter.kt
package de.markusfisch.android.shadereditor.adapter

import android.content.Context
import android.database.Cursor
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SpinnerAdapter
import de.markusfisch.android.shadereditor.R

class ShaderSpinnerAdapter(
    context: Context, cursor: Cursor
) : ShaderAdapter(context, cursor), SpinnerAdapter {

    override fun newDropDownView(context: Context, cursor: Cursor, parent: ViewGroup): View {
        return LayoutInflater.from(parent.context).inflate(R.layout.row_shader, parent, false)
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
        setData(getViewHolder(view), cursor)
    }
}
